docs/tips.rst

```diff
@@ -110,6 +123,98 @@ branch of your GitHub repository::
 
 (assuming origin is set to your remote repository).
 
+Publishing to GitHub Pages Using a Custom GitHub Actions Workflow
+-----------------------------------------------------------------
+
+Pelican comes with a `custom workflow <https://github.com/getpelican/pelican/blob/master/.github/workflows/github_pages.yml>`_
+for publishing a Pelican site. To use it:
+
+1. Enable GitHub Pages in your repo: go to **Settings → Pages** and choose
+   **GitHub Actions** for the **Source** setting.
+
+2. Commit a ``.github/workflows/pelican.yml`` file to your repo with these contents:
+
+   .. code-block:: yaml
+
+       name: Deploy to GitHub Pages
+       on:
+         push:
+           branches: ["main"]
+         workflow_dispatch:
+       jobs:
+         deploy:
+           uses: "getpelican/pelican/.github/workflows/github_pages.yml@main"
+           permissions:
+             contents: "read"
+             pages: "write"
+             id-token: "write"
+
+3. Go to the **Actions** tab in your repo
+   (``https://github.com/<username>/<repository>/actions``) and you should see a
+   **Deploy to GitHub Pages** action running.
+
+4. Once the action completes you should see your Pelican site deployed at your
+   repo's GitHub Pages URL: ``https://<username>.github.io`` for a user or
+   organization site or ``https://<username>.github.io/<repository>>`` for a
+   project site.
+
+Notes:
+
+* Your Pelican settings file should be named ``publishconf.py`` in the root of
+  your repo. If you want to use a different settings file pass a ``settings``
+  option to the workflow:
+
+   .. code-block:: yaml
+
+       deploy:
+         uses: getpelican/pelican/.github/workflows/github_pages.yml@main
+         permissions:
+           contents: "read"
+           pages: "write"
+           id-token: "write"
+         with:
+           settings: "my_settings.py"
+
+* If you've changed your Pelican ``OUTPUT_PATH`` setting from the default you
+  should pass an ``output-path`` setting to tell GitHub Actions about your
+  custom output path:
+
+   .. code-block:: yaml
+
+       with:
+         output-path: "__output__/"
```

Comment:
It might be possible for the workflow to automatically determine the value of this and other settings from the output of `pelican --print-settings` but I found this awkward to implement: I thought I could just do something like `pelican --print-settings | jq '.OUTPUT_PATH'` but the output of `--print-settings` doesn't appear to be valid JSON (?) and pelican also prints warnings at the top of the output even with `--quiet`.

Reply:
Ah, the workflow should just use `--output` to force the output path to a known value

Reply:
Done